In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

pd.set_option("display.max_columns", 100)

DATA_PATH = "data/Moscow_cleaned.csv.gz"

df = pd.read_csv(DATA_PATH)

df.shape

(2044405, 14)

In [2]:
df.head()

,request_year,request_region,id,url,title,price,price_metric,source,city,city1,metro,address,person_type,publication_time
0,2023,Москва,592510619,https://www.cian.ru/sale/flat/281955840,"3-к кв. Москва ул. Казакова, 7 (164.7 м²)",101479168,NaN,cian.ru,Москва,Москва,Курская,"ул. Казакова, 7",Агентство,2023-01-01 00:30:18
1,2023,Москва,592512481,https://www.cian.ru/sale/flat/281956494,"1-к кв. Москва Волоколамское ш., 71к9 (35.0 м²)",10629570,NaN,cian.ru,Москва,Москва,Спартак,"Волоколамское ш., 71к9",Агентство,2023-01-01 00:30:19
2,2023,Москва,592537336,https://www.cian.ru/sale/flat/281976963,"3-к кв. Москва Куркинское ш., 17 (96.8 м²)",24800000,NaN,cian.ru,Москва,Москва,СЗАО,"Куркинское ш., 17",Частное лицо (фильтр),2023-01-01 00:30:23
3,2023,Москва,591008856,https://www.cian.ru/sale/commercial/280944356,"Торговая площадь в Москва ул. Остоженка, 1/9 (...",270000000,NaN,cian.ru,Москва,Москва,Кропоткинская,"ул. Остоженка, 1/9",Агентство,2023-01-01 00:49:37
4,2023,Москва,591014495,https://www.cian.ru/sale/commercial/280948323,Помещение свободного назначения в Москва Огоро...,283493400,NaN,cian.ru,Москва,Москва,Бутырская,"Огородный проезд, 5",Агентство,2023-01-01 00:49:37


In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2044405 entries, 0 to 2044404
Data columns (total 14 columns):
 #   Column            Dtype
---  ------            -----
 0   request_year      int64
 1   request_region    str  
 2   id                int64
 3   url               str  
 4   title             str  
 5   price             int64
 6   price_metric      str  
 7   source            str  
 8   city              str  
 9   city1             str  
 10  metro             str  
 11  address           str  
 12  person_type       str  
 13  publication_time  str  
dtypes: int64(3), str(11)
memory usage: 218.4 MB


In [4]:
df.isna().mean().sort_values(ascending=False)

price_metric        0.011619
address             0.005512
metro               0.000222
request_year        0.000000
request_region      0.000000
id                  0.000000
url                 0.000000
title               0.000000
price               0.000000
source              0.000000
city                0.000000
city1               0.000000
person_type         0.000000
publication_time    0.000000
dtype: float64

## Найдены пропущенные значения

В датасете пропуски есть только в трех колонках:

- `price_metric` - около 1.16%;
- `address` - около 0.55%;
- `metro` - около 0.02%.

Критически важных пропусков в основных полях нет: `price`, `title`, `id`, `source`, `person_type`, `publication_time` заполнены полностью.

Колонка `price_metric` не будет использоваться как исходный признак, потому что цену за квадратный метр надежнее рассчитать самостоятельно после извлечения площади из `title`.

Пропуски в `address` и `metro` составляют малую долю данных. Чтобы не терять наблюдения, значения будут заполнены категорией `Unknown`.

In [8]:
df_clean = df.copy()

df_clean["address"] = df_clean["address"].fillna("Unknown")
df_clean["metro"] = df_clean["metro"].fillna("Unknown")

df_clean = df_clean.drop(columns=["price_metric"])

In [13]:
df_clean.isna().mean().sort_values(ascending=False)

request_year        0.0
request_region      0.0
id                  0.0
url                 0.0
title               0.0
price               0.0
source              0.0
city                0.0
city1               0.0
metro               0.0
address             0.0
person_type         0.0
publication_time    0.0
dtype: float64

In [12]:
df.duplicated("id").sum()

np.int64(0)

In [15]:
df_clean.to_csv("data/moscow_clean_base.csv.gz", index=False)